packages/kit/types/private.d.ts

```diff
@@ -12,8 +12,11 @@ export interface AdapterEntry {
 
 	/**
 	 * A function that compares the candidate route with the current route to determine
-	 * if it should be treated as a fallback for the current route. For example, `/foo/[c]`
-	 * is a fallback for `/foo/a-[b]`, and `/[...catchall]` is a fallback for all routes
+	 * if it should be grouped with the current route.
+	 *
+	 * Use cases:
+	 * - Fallback pages: `/foo/[c]` is a fallback for `/foo/a-[b]`, and `/[...catchall]` is a fallback for all routes
+	 * - Grouping routes that share a common `config`: `/foo` should be deployed to the edge, `/bar` and `/baz` should be deployed to a serverless function
 	 */
 	filter(route: RouteDefinition): boolean;
```

Comment:
Found that `filter` is usable to create groups of routes for a specific function. Changed the docs to reflect that. I feel like `group` would have been a better name - worth to keep in mind for 2.0?

Reply:
Yeah. We could also rename it now but keep `filter` with a deprecation warning, though it might be worth waiting until route level config has been around for a bit longer to see if its usage turns up any other surprises

Reply:
I just realized that `filter` does not do what I want - we need a new `group` function.
- `filter` checks all lower priority routes if they should be moved into the entry, but if some should, those are not removed from the rest of the routes to be looked at
- `group` needs to sort every route into a bucket

Reply:
It's a mess trying to tuck `group` onto the existing `createEntries`. `filter` has two conflicting requirements:
1. it shouldn't have any effect when groups exist
2. it should add everything to one when split false

To find out if groups exist, you first need to traverse all facades.
It's confusing both for the devs using it and us framework authors implementing it.